ecr/base.go

```diff
@@ -108,19 +108,18 @@ func (b *ecrBase) runGetImage(ctx context.Context, batchGetImageInput ecr.BatchG
 	}
 	log.G(ctx).WithField("batchGetImageOutput", batchGetImageOutput).Trace("ecr.base.image")
 
-	if len(batchGetImageOutput.Images) != 1 {
-		for _, failure := range batchGetImageOutput.Failures {
-			switch aws.StringValue(failure.FailureCode) {
-			case ecr.ImageFailureCodeImageTagDoesNotMatchDigest:
-				log.G(ctx).WithField("failure", failure).Debug("ecr.base.image: no matching image with specified digest")
-				return nil, errImageNotFound
-			case ecr.ImageFailureCodeImageNotFound:
-				log.G(ctx).WithField("failure", failure).Debug("ecr.base.image: no image found")
-				return nil, errImageNotFound
-			}
+	for _, failure := range batchGetImageOutput.Failures {
+		switch aws.StringValue(failure.FailureCode) {
+		case ecr.ImageFailureCodeImageTagDoesNotMatchDigest:
+			log.G(ctx).WithField("failure", failure).Debug("ecr.base.image: no matching image with specified digest")
+			return nil, errImageNotFound
+		case ecr.ImageFailureCodeImageNotFound:
+			log.G(ctx).WithField("failure", failure).Debug("ecr.base.image: no image found")
+			return nil, errImageNotFound
+		default:
+			log.G(ctx).WithField("failure", failure).Warn("ecr.base.image: unhandled request failure")
+			return nil, reference.ErrInvalid
```

Comment:
So this is a behavior change: instead of iterating through all errors and trying to find one that matches, we just look at the first error.  (We don't even need the for loop right now.)  Is that the behavior that we want?

Reply:
As with another comment - this became unnecessary over the course of development. It's reworked (it's much closer to the original) and should be reflected in the latest push.